src/tools/cephfs/cephfs-shell

```diff
@@ -733,6 +733,11 @@ exists.')
                     self.poutput(path, end=':\n')
                 items = sorted(ls(path),
                                key=lambda item: item.d_name)
+            if not items:
```

Comment:
> In case of empty directory None is returned and with non-existing directory empty list is returned.

Keep in mind:

````
$ python3
Python 3.7.3 (default, Jun 24 2019, 04:54:02) 
[GCC 9.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> not []
True
>>> not None
True
````

Reply:
So an empty directory would be seen as not existing?

Reply:
> > In case of empty directory None is returned and with non-existing directory empty list is returned.
> 
> Keep in mind:
> 
> ```
> $ python3
> Python 3.7.3 (default, Jun 24 2019, 04:54:02) 
> [GCC 9.1.0] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> not []
> True
> >>> not None
> True
> ```

Thank you for pointing out, I have updated the patch.

Reply:
> So an empty directory would be seen as not existing?

No, for empty directory nothing would be printed. Error message is only printed for non existing directories.